xrmocap/core/estimation/mview_mperson_topdown_estimator.py

```diff
@@ -195,66 +216,130 @@ def run(
+                    associate_idxs = sframe_association_results[p_idx]
+                    tri_kps2d = np.zeros((n_view, n_kps, 3))
+                    tri_mask = np.zeros((n_view, n_kps, 1))
                     for view_idx in range(n_view):
                         kps2d_idx = associate_idxs[view_idx]
-                        if kps2d_idx is not None:
-                            tri_kps2d[view_idx] = \
-                                sframe_keypoints2d_list[
-                                    view_idx].get_keypoints()[
-                                        0, kps2d_idx, ...]
-                            tri_mask[view_idx] = \
-                                sframe_keypoints2d_list[
-                                    view_idx].get_mask()[
-                                        0, kps2d_idx, ...]
+                        if not np.isnan(kps2d_idx):
+                            tri_kps2d[view_idx] = sframe_keypoints2d_list[
+                                view_idx].get_keypoints()[0, int(kps2d_idx)]
+                            tri_mask[view_idx, :, 0] = sframe_keypoints2d_list[
+                                view_idx].get_mask()[0, int(kps2d_idx)]
                     if self.point_selectors is not None:
                         for selector in self.point_selectors:
                             tri_mask = selector.get_selection_mask(
                                 points=tri_kps2d, init_points_mask=tri_mask)
+                    # Triangulation
                     kps3d = self.triangulator.triangulate(tri_kps2d, tri_mask)
-                    max_identity = max(max_identity, identity)
-                    # TODO: set kps3d to proper location by identity
-        keypoints3d_list = [
-            Keypoints(kps=kps3d, mask=np.ones_like(kps3d[..., 0]))
-        ]
-        smpl_data_list = [
-            None,
-        ] * len(keypoints3d_list)
-        for person_idx, keypoints3d in enumerate(keypoints3d_list):
-            keypoints3d = self.optimize_keypoints3d(keypoints3d)
-            smpl_data = self.estimate_smpl(keypoints3d)
-            keypoints3d_list[person_idx] = keypoints3d
-            smpl_data_list[person_idx] = smpl_data
-        return keypoints3d_list, smpl_data_list
+                    if identity > max_identity:
+                        n_identity = identity - max_identity
+                        pred_kps3d = np.concatenate(
+                            (pred_kps3d,
+                             np.zeros((n_frame, n_identity, n_kps, 4))),
+                            axis=1)
+                        max_identity = identity
+                    pred_kps3d[start_idx + sub_fidx,
+                               identity] = np.concatenate(
+                                   (kps3d, np.ones_like(kps3d[:, 0:1])),
+                                   axis=-1)
+                for i in range(max_identity + 1):
+                    if i in identities:
+                        index = identities.index(i)
```

Comment:
These 3 lines can be replaced by:
```python
for index, i in enumerate(identities):
    ...
```
Think twice about var names, it could be:
```python
for p_idx, identity in enumerate(identities):
    ...
```